File: ui/src/app/shared/activity.ts
export enum ActionType {
  Create = 'Create',
  Update = 'Update',
  Delete = 'Delete',
  Login = 'Login',
  Logout = 'Logout'
}

export interface Activity {
  id: string;
  type: ActionType;
  contentType: string;
  whenCreated: Date;
  userId: string;
  userName: string;
  userUsername: string;
  projectId: string;
  projectName: string;
  before: string;
  after: string;
}

File: src/java/dto/Activity.java
package dto;

import org.joda.time.DateTime;

import java.util.UUID;

public class Activity extends Dto {
  public UUID id;
  public ActionType type;
  public String contentType;
  public DateTime whenCreated;
  public UUID userId;
  public String userName;
  public String userUsername;
  public String userEmailHash;
  public UUID projectId;
  public String projectName;
  public String before;
  public String after;
}
